OneSignalSDK/onesignal/in-app-messages/src/main/java/com/onesignal/inAppMessages/internal/triggers/impl/DynamicTriggerController.kt

```diff
@@ -44,14 +44,31 @@ internal class DynamicTriggerController(
             }
 
             var currentTimeInterval: Long = 0
+            val lastTimeAppDismissed = _state.lastTimeInAppDismissed
             when (trigger.kind) {
-                Trigger.OSTriggerKind.SESSION_TIME ->
+                Trigger.OSTriggerKind.SESSION_TIME -> {
                     currentTimeInterval = _time.currentTimeMillis - _session.startTime
+                    // if last time IAM dismissed also satisfied this trigger, do not fire it again
+                    val lastTimeDismissedInterval =
+                        if (lastTimeAppDismissed == null) {
+                            0
+                        } else {
+                            _time.currentTimeMillis - lastTimeAppDismissed
+                        }
+                    if (evaluateTimeIntervalWithOperator(
+                            lastTimeDismissedInterval.toDouble(),
+                            currentTimeInterval.toDouble(),
+                            trigger.operatorType,
+                        )
+                    ) {
+                        return false
+                    }
```

Comment:
Won't this break the "Every time trigger conditions are satisfied" if the IAM had a custom trigger with AND?

Reply:
Yes it does. I pushed a new approach with memorization.